src/main/java/io/jenkins/plugins/uleska/api/HttpScanApi.java

```diff
@@ -0,0 +1,73 @@
+package io.jenkins.plugins.uleska.api;
+
+import hudson.model.TaskListener;
+import org.apache.hc.client5.http.classic.HttpClient;
+import org.apache.hc.client5.http.classic.methods.HttpGet;
+import org.apache.hc.core5.http.HttpResponse;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.UUID;
+
+public class HttpScanApi implements ScanApi {
+
+    private static final String SCAN_ADDRESS = "%s/SecureDesigner/api/v1/applications/%s/versions/%s/scan";
+
+    private final TaskListener taskListener;
+    private final HttpFactory httpFactory;
+    private final String host;
+    private final char[] apiKey;
+
+    public HttpScanApi(TaskListener taskListener, HttpFactory httpFactory, String host, char[] apiKey) {
+        this.httpFactory = httpFactory;
+        this.host = host;
+        this.apiKey = apiKey;
+        this.taskListener = taskListener;
+    }
+
+    @Override
+    public void doScan(UUID applicationId, UUID versionId) throws ScanException {
+        String address = String.format(SCAN_ADDRESS, host, applicationId, versionId);
+        doHttpGet(address);
+    }
+
+    @Override
+    public void close() throws Exception {
```

Comment:
This method doesn't need "throws exception"

Reply:
Changed Thanks